libbeat/scripts/cmd/global_fields/main.go

```diff
@@ -27,37 +27,55 @@ import (
 )
 
 func main() {
-	esBeatsPath := flag.String("es_beats_path", "..", "Path to elastic/beats")
-	beatPath := flag.String("beat_path", ".", "Path to your Beat")
+	var (
+		esBeatsPath string
+		beatPath    string
+		output      string
+	)
+	flag.StringVar(&esBeatsPath, "es_beats_path", "..", "Path to elastic/beats")
+	flag.StringVar(&beatPath, "beat_path", ".", "Path to your Beat")
+	flag.StringVar(&output, "out", "-", "Path to output. Default: stdout")
 	flag.Parse()
 
 	beatFieldsPaths := flag.Args()
-	name := filepath.Base(*beatPath)
+	name := filepath.Base(beatPath)
 
-	if *beatPath == "" {
+	if beatPath == "" {
 		fmt.Fprintf(os.Stderr, "beat_path cannot be empty")
 		os.Exit(1)
 	}
 
-	err := os.MkdirAll(filepath.Join(*beatPath, "_meta"), 0755)
+	esBeats, err := os.Open(esBeatsPath)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Cannot create _meta dir for %s: %+v\n", name, err)
+		fmt.Fprintf(os.Stderr, "Error opening elastic/beats: %+v\n", err)
+		os.Exit(1)
+	}
+	beat, err := os.Open(beatPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error opening target Beat: %+v\n", err)
+		os.Exit(1)
+	}
+	esBeatsInfo, err := esBeats.Stat()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting file info of elastic/beats: %+v\n", err)
+		os.Exit(1)
+	}
+	beatInfo, err := beat.Stat()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting file info of target Beat: %+v\n", err)
 		os.Exit(1)
 	}
 
-	if len(beatFieldsPaths) == 0 {
-		fmt.Println("No field files to collect")
-		err = fields.AppendFromLibbeat(*esBeatsPath, *beatPath)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Cannot generate global fields.yml for %s: %+v\n", name, err)
-			os.Exit(2)
+	if len(beatFieldsPaths) == 0 && os.SameFile(esBeatsInfo, beatInfo) {
```

Comment:
Trying to follow what this does. Could you quickly elaborate?

Reply:
If a community Beat does not have its own`fields.yml` file, it still requires the fields coming from libbeat to generate e.g assets. In case of Elastic Beats, it's not a problem because all of them has unique fields.yml files somewhere.

Reply:
Thanks for the details. Could you perhaps add a comment to the line to make sure if we refactor this in the future we don't get rid of it?